entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -94,10 +113,17 @@ class EntityHandler(
      */
     @GetMapping("/{entityId}", produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
     fun getByURI(@PathVariable entityId: String): Mono<ResponseEntity<*>> {
-        return entityId.toMono()
-            .map {
+        return entityId.toMono().zipWith(extractSubjectOrEmpty())
+            .doOnNext {
                 if (!entityService.exists(entityId)) throw ResourceNotFoundException("Entity Not Found")
-                entityService.getFullEntityById(it)
+                if (!authorizationService.userHasReadRightsOnEntity(

```

Comment:
For simplicity purposes I did a method contract easier to read/understand for this part. This can of course be changed. But if you check the implementation I basically do what you suggest, just a lower layer.
What do you think, should I remove the added methods ?

Reply:
I got it, okay, it's up to you to decide.